ios/MaasCore/Sources/MaasComponents/Supporting Files/ParseColors.swift

```diff
@@ -0,0 +1,32 @@
+public extension String {
+    func parseColor() -> Color {
+        .init(hex: self)
+    }
+}
+
+public extension Color {
+    init(hex: String) {
+        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
+        var int: UInt64 = 0
+        Scanner(string: hex).scanHexInt64(&int)
+        let a, r, g, b: UInt64
+        switch hex.count {
+        case 3: // RGB (12-bit)
+            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
+        case 6: // RGB (24-bit)
+            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
+        case 8: // ARGB (32-bit)
+            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
+        default:
+            (a, r, g, b) = (1, 1, 1, 0)
```

Comment:
a strange choice for default to be a `yellow` color 😅 
```suggestion
            (a, r, g, b) = (0, 1, 1, 1)
```

Reply:
Okay :D